### PR TITLE
PAT-319 - Connect Postgres DB to hello-cammis-data

### PR DESCRIPTION
**Pull request by gubbin**

### CODE REVIEW

File: data/views.py [hunk 1]
```diff
@@ -40,8 +40,8 @@ def ready(request):
 def ready_database():
     # Save a record to the database
     unlikely_name = "jkadfsjk3jl2"
-    if controller.add_greeting(unlikely_name, "Ready", "!") == None:
+    if controller.add_greeting(unlikely_name, "Ready") == None:
```
Comment (gubbin) [inline data/views.py]: Little bugfix

File: helm/hello-cammis-data/templates/deployment.yaml [hunk 2]
```diff
@@ -11,6 +11,8 @@ spec:
       app: {{ .Values.global.helloCammisData.name }}
   template:
     metadata:
+      annotations:
```
Comment (gubbin) [inline helm/hello-cammis-data/templates/deployment.yaml]: This forces a new pod if the DB config changes - see PAT-359

File: helm/hello-cammis-data/values.yaml [hunk 3]
```diff
@@ -5,3 +5,14 @@ global:
         name: hello-cammis-data
         imageName: 166636506311.dkr.ecr.us-west-2.amazonaws.com/hello-cammis-data
         # imageVersion: 0.0.0 # This can be use via command line `--set global.helloCammisData.imageVersion=0.0.0`, otherwise default is `version-{{ .Chart.Version }}`
+        dbEngine: postgresql_psycopg2  # default is 'sqlite'
+        dbHost: review-env-postgresql  # Default is 'BASE_DIR/db.sqlite3'
+        # was: 'host.docker.internal' - localdev?
+        djangoDebug: true  # Run Django in debug mode
+        addTestData: true  # Load test data into DB on install
```
Comment (gubbin) [inline helm/hello-cammis-data/values.yaml]: Set to "false" on environments with existing data such as UAT or Production

File: helm/hello-cammis-data/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  DJANGO_APP_DB_ENGINE: '{{ .Values.global.helloCammisData.dbEngine | default "sqlite3" }}'
+  DJANGO_APP_DB_HOST: '{{ .Values.global.helloCammisData.dbHost | default "" }}'
+  DJANGO_APP_DB_PORT: '{{ .Values.global.postgresql.servicePort | default "" }}'
+  DJANGO_APP_DB_NAME: '{{ .Values.global.postgresql.postgresqlDatabase | default "" }}'
+  DJANGO_APP_DB_USER: '{{ .Values.global.postgresql.postgresqlUsername | default "" }}'
+  DJANGO_APP_DB_PASSWORD: '{{ .Values.global.postgresql.postgresqlPassword | default "" }}'
+  DJANGO_DEBUG: '{{ .Values.global.helloCammisData.djangoDebug | default true | ternary "TRUE" "FALSE" }}'
+  DJANGO_SECRET_KEY: '{{ .Values.global.helloCammisData.secretKey | default "abcd1234" }}'
+---
+# Environment for Django's settings.py - DB admin
```
Comment (gubbin) [inline helm/hello-cammis-data/templates/configmap.yaml]: Yes, this is a duplicate, but I wanted to get this plumbed-in. Hoping the FDR team uses the same pattern.

File: helm/hello-cammis-data/templates/databaseJobs.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: hello-cammis-data-migrate
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-weight": "4"
+    "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (gubbin) [inline helm/hello-cammis-data/templates/databaseJobs.yaml]: Ensures the job runs on subsequent helm deploys

File: helm/hello-cammis-data/templates/databaseJobs.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+spec:
+  template:
+    spec:
+      initContainers:
+        - name: wait-for-postgresql
```
Comment (gubbin) [inline helm/hello-cammis-data/templates/databaseJobs.yaml]: Gotta wait for the DB server to be ready!

**Comment by krenshaw2018** [comment]
Done, merged, and closed.

**Pull request merged**